client/src/context/useNotificationContext.tsx

```diff
@@ -0,0 +1,37 @@
+import { Notification } from '../interface/Notification';
+import { fetchNotifications } from '../helpers/APICalls/getNotifications';
+import { updateNotifications } from '../helpers/APICalls/updateNotifications';
+import { createContext, FunctionComponent, useState, useContext, useEffect } from 'react';
+
+interface INotificationContext {
+  notifications: [Notification] | null | undefined;
+  update: () => void;
+}
+
+export const NotificationContext = createContext<INotificationContext>({
+  notifications: null,
+  update: () => null,
+});
+
+export const NotificationProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [notifications, readNotifications] = useState<[Notification] | null | undefined>();
```

Comment:
Same concern about the `set...` naming convention here. In this case, "read" is also misleading since you don't necessarily mark all fetched notifications as read (the user needs to click on the menu).

Reply:
Applied. 'read' is confusing as it can be a verb or an adjective.